src/python/pants/invalidation/build_invalidator.py

```diff
@@ -31,11 +25,23 @@
 GLOBAL_CACHE_KEY_GEN_VERSION = '7'
 
 
-class CacheKeyGenerator(object):
-  """Generates cache keys for versions of target sets."""
+class CacheKey(datatype('CacheKey', ['id', 'hash'])):
```

Comment:
The switch here to `datatype` from `namedtuple` exposed a bug in https://github.com/pantsbuild/pants/commit/46c9fc3ddcc9ed6adb2ce3e204fd5e6c37d1ecac#commitcomment-24581886: https://travis-ci.org/pantsbuild/pants/jobs/280082515
Fix coming

Reply:
Hrm, there is enough code I don't understand, or that is broken and/or obsolete here that the most prudent thing to do seems to be to revert to namedtuple to preserve long existing behavior unrelated to this rb.